### ISSUE TITLE
Suggestion: [Org / PM] Meta Repositories to support & encourage better PM

### ISSUE THREAD

**Issue by DennisSmolek** [pmndrs/docs#231]
# Problem:

- Right now the organization has many projects in rapid active development, but no place to converse, track, share, or manage things.

- This lack of structure and oversight is leading to a general state of confusion, even amongst very well intentioned, active contributors and managers. 

- Users struggle to engage with the project and don’t benefit from cross communication/standards
- No clear space for Public "Clean" repository and "Messy" working dev space/notes, guides, outlines. Everyone uses their own thing..

**<details><summary> **Effected Departments/Orgs/Teams** </summary>**
- Tasks
- Schedules (Calendar)
- Releases
  - Milestones
  - Project Milestones
- Documentation
- Deployment
- Design Systems
- Repo Issues/bugs
- Responsibilities
  - Ownership
  - Decision Authority
- Goals
- Roadmaps
- Etc.
</details>

## TLDR Solution:

Create “meta” repositories to act as ***DevGroup*** boxes

Create Project Templates for Project Owners

### **The Paradox of Choice**:

There are simply too many ways, tools, preferences, services, ideologies to do things. Everything integrates with github but no one knows what the “right way” is to do anything. **And most things are FREE**

### **The *Obvious* Decision**:

- Everything is on Github, and as public project nearly all github features are free and supported.
- The org can roll out/add structures and support as the Project Manager needs/asks.
- Github themselves pushes the uni-repo mentality and framework heavily.

### Why that doesn’t actually work:

Public repositories are treated as up-to-date snapshots of the life of the project. Most devs **IMMEDIATELY** look at a few things: 

- Number of Open Issues
- Last Commit to main
- last Release
- Header of the Readme.

Github knows this:
<div >
<img width="250" alt="Screen Shot 2022-02-15 at 03 36 50" src="https://user-images.githubusercontent.com/1397052/154050705-5913ffcd-ea00-49b7-ac56-f7072f3c2587.png">
<img width="250" alt="Screen Shot 2022-02-15 at 03 34 54" src="https://user-images.githubusercontent.com/1397052/154050712-b3e0e9a4-7ab1-495b-b5e4-3bf35448cb59.png">
</div>
---

*However...*

## GITHUB PROJECTS TRACKS ALL ACTIVITIES AS ISSUES

This means every small task, request, etc adds to the repo issue count. Long running discussions, or super issues (holder of others) gets seen as backlog/bad management
<details><summary>A Few more pains</summary>
Dev’s cant have freedom to discuss options because any concept on the main repo issue becomes “the devs real opinion”

There are many downsides I can list later if needed**
</details>

## Consider writing docs for all the functions/methods in the codebase. 
You could automate a task to use JSDoc and use a template to create the project management for the function and its docs page creation. 

In a PM tool that would look something like:
<img width="600" alt="Screen Shot 2022-02-15 at 04 34 29" src="https://user-images.githubusercontent.com/1397052/154052018-dce373b5-9bad-4975-b740-a47c408c0c3e.png">

## What are people doing about it?

In the 3 groups I’m working with and in all the channels I’ve seen its usually
1. everyone forks the repo...

2. . . .

3. Profit....

_<details><summary>But seriously,..._</summary>

-  the “dev box” of projects is going on private peoples forked copies, and OFTEN not the project manager/owner. 

- This causes A LOT of hesitancy/uncertainty for people to know how to contribute. 

- No one knows who’s responsible for what or what the status is for anything

- there are no milestones

- There are no outlines

- goals aren’t clearly defined...

- REALLY amazing documentation is in THE WEIRDEST place

(The best outline and explanation for Leva’s plugin system including skeleton examples, is in the Type definition file.... It’s better than the readme and example project has no docs)
</details>

# Proposed Solution

### Use Github Projects (Beta) with custom views

Projects are no longer repo specific, issues require a repo to live on

### Create a general pmndrs/DevGroup repo. & Project

(***Highest Role restrictions)***
- Central Hub for all the entire org
<details><summary>See Full List...</summary>

- This can be the future house of the design system/guide
- Can be the home for the website template/system
- General Org dev standards can go here (code conventions, semvar, deployment)
- Assets for other projects ( Like the Vike sandboxes, deployments to other guides, or assets from partnerships) can be stored & Managed centrally
- Issues, Milestones, etc dont matter.

</details>

### Create a DevGroup or “meta” or whatever name repo for each project
(lower ***Role restrictions)***
- Easy to point to central place for each project
- Project Managers have direct permissions to add/remove contributors

<details><summary>See Full List...</summary>
- Issue count doesnt matter
- KEY POINT. Dev boxes are  temporary.***
- Ideally between milestones which should be releases  or a feature, this keeps the main repo the center of truth
- Give contributors higher level access
- They feel more important & invloved, & RESPONSIBLE
- contributors can close issues
- Owner can put resources on it and they can PR to it, but a
- Milestones dont have to directly relate to the codebase.. Can be a feature set, or some other goals
- Users don’t get confused about those milestones or why issues are left open for a year
- I can go on and on

</details>

## Option 2

- Create a DevResorces repo as in Option 1

- Create a general DevGroup repo to hold all issues/pm related things 

### Reason I don’t like this as much:

In the projects you would have to add tags to the issue to see them on a per-project basis
This means every small task would have to manually get the task added
this will add up, and people wont make tasks...
Too much in one space

# "Ok so what do we need today  a big outline or something?!?"

No, no reason to make it drawn out or crazy. 
1. **Figure out the names.** In the past I'd call it "Meta" because its still all public but Facebook ruined that word for everyone..
 "DevGroup" seems ok
2.  Create pmndrs/DevGroup. use it for shared stuff
3. Create a project for the org that's admin/owners
4. Then any org project can be made and assigned
5.  Create 'DevGroup' repos for the projects with teams, put a issue in the project
6.  Create a Generic Project for that repo
7. Assign the Project Manager as a Admin so they can invite other members to their "dev group"

8. Let them work the rest out

Why not ask them first?
People are weird, and theres so much going on if you ask them to set something up they probably wont or wont get around to it
Make the repo, make them an owner with the project.
If they REALLY dont want to be organized or have some other method, they can request it be deleted..

** You can/will have to create public projects, but I think project members have to be inited **

Wow this was a brain dump.. LMK what you think and if you have a different opinion or better solution! This is just what's been bouncing around in my head

**Comment by gsimone** [comment]
1. generally sounds like a good plan
2. working group is the usual term i see used around, would that work?

**Comment by stephencorwin** [comment]
Interesting take. I definitely agree that milestones and roadmaps are useful.
In terms of "no place to converse", we have a very active discord.  
  
[![Discord Shield](https://discordapp.com/api/guilds/740090768164651008/widget.png?style=banner2)](https://discord.gg/poimandres)

**Comment by abernier** [comment]
I close this issue since this repo is just about documentation generator, no more.
